typescript/infra/src/utils/gcloud.ts

```diff
@@ -9,16 +9,49 @@ interface IamCondition {
   expression: string;
 }
 
+// Allows secrets to be overridden via environment variables to avoid
+// gcloud calls. This is particularly useful for running commands in k8s,
+// where we can use external-secrets to fetch secrets from GCP secret manager,
+// and don't necessarily want to use gcloud from within k8s.
+// See tryGCPSecretFromEnvVariable for details on how to override via environment
+// variables.
 export async function fetchGCPSecret(secretName: string, parseJson = true) {
-  const [output] = await execCmd(
-    `gcloud secrets versions access latest --secret ${secretName}`,
-  );
+  let output: string;
+
+  const envVarOverride = tryGCPSecretFromEnvVariable(secretName);
+  if (envVarOverride !== undefined) {
+    console.log(
+      `Using environment variable instead of GCP secret with name ${secretName}`,
+    );
+    output = envVarOverride;
+  } else {
+    [output] = await execCmd(
+      `gcloud secrets versions access latest --secret ${secretName}`,
+    );
+  }
+
   if (parseJson) {
     return JSON.parse(output);
   }
   return output;
 }
 
+// If the environment variable GCP_SECRET_OVERRIDES_ENABLED is `true`,
+// this will attempt to find an environment variable of the form:
+//  `GCP_SECRET_OVERRIDE_${gcpSecretName..replaceAll('-', '_').toUpperCase()}`
+// If found, it's returned, otherwise, undefined is returned.
+function tryGCPSecretFromEnvVariable(gcpSecretName: string) {
+  const overridingEnabled =
+    process.env.GCP_SECRET_OVERRIDES_ENABLED?.toLowerCase().trim() === 'true';
```

Comment:
It would probably be sufficient to just check that `GCP_SECRET_OVERRIDES_ENABLED` has any value at all. E.g. if it's `1` still want to treat that as true